### DISCUSSION TITLE
Is "multi-segment language detection" based on accent or actually spoken speech?

### DISCUSSION THREAD

**Discussion by toanhn** [SYSTRAN/faster-whisper#1035] [Q&A]
When I use "multi-segment language detection" https://github.com/SYSTRAN/faster-whisper?tab=readme-ov-file#multi-segment-language-detection.
I found that it detects language based on accent rather than actual spoken speech.
Ex: I have an audio file, a speaker is Malaysian and he says English, if I use "native language detection" with the first 30 seconds of the audio file then it detects the speech is in English. But when I use "multi-segment language detection" then it detects the speech is in Malay.

Does anyone have any idea on this issue? Thanks.

**Comment by Hikari-Tsai** [comment]
Whisper language detection predicts the language token from the audio representation; it is not an accent classifier. However, accent, speaker identity, topic and recording domain are all present in that representation and can bias the prediction—especially for closely related languages and code-switched speech.

There are two different modes involved:

- initial/native detection chooses one language from the beginning of the input;
- `multilingual=True` repeats detection for each ~30-second transcription window and changes the tokenizer language for that window.

The second mode has less context per decision, so a Malaysian-accented English window can be classified as Malay even though the lexical content is English. That is a model error, not evidence that the algorithm intentionally detects accent.

If the recording is known to be English, pass `language="en"`. If it can switch languages, use `multilingual=True` but validate the output; for reliable language labels, use a dedicated spoken-language-identification model on sufficiently long speech regions. Avoid making decisions from very short windows, silence, or background music.
